### ISSUE TITLE
Update Required

### ISSUE THREAD

**Issue by ElReyZero** [polarathene/biglobby#63]
I want to play with this mod, please update it to BLT 2.0

**Comment by polarathene** [comment]
Hi there, this project is very low prority for me these days. I have other personal projects and work projects that take precedence. I've not played PD2 in quite some time, nor see much value in continuing to maintain this mod in my free time.

I welcome contributions. You can encourage others to do those or myself by building up a bounty for the task with others interested [here](https://www.bountysource.com/issues/49602212-update-required).

**For contributors:**
I am happy to answer questions as best I can for anyone wanting to contribute an update. Beyond making changes to support BLT 2.0, it should mostly just be a a sync update(see past commit history). Although I believe Overkill were planning some features that may break the compatibility of this mod in future requiring more time investment to resolve.

**Comment by polarathene** [comment]
A bounty has been posted by a community member. Feel free to pitch in to see the mod updated by supporting the bounty [here](https://www.bountysource.com/issues/49602212-update-required).

Just to be clear to any dev taking on this bounty. The bounty is intended for a working mod update(transition to BLT 2.0, network methods update(pdmod), anything else required to have the mod in a working state such as checking for code additions to existing overrides or UI/hardcoded fixes like those that have already been implemented). 

It's been months since the last update to this mod was made so it may not go as smoothly now depending how much the game has changed. Best of luck :)

**Comment by MegaBaconSticks** [comment]
Hi Polarathene! I was wondering if I could possibly work on the mod to be up to date with BLT 2.0, or make a standalone remastered version with cleaner code for Big Lobby. As for the moment right now of being an 8th Grader, I want to accept this as a challenge to actually accomplish something when it comes to coding. But, I really want to ask badly for what do you recommend for educational sources of coding with languages like Lua, or what sources have you possibly used for the development of Big Lobby? Thanks!

**Comment by polarathene** [comment]
Hi @MegaBaconSticks , my advice is the same as I [provided to you on my reddit account](https://www.reddit.com/r/paydaytheheistmods/comments/7fcv7p/anyone_know_how_to_update_the_big_lobby_mod_to/dqf17io/?context=3).

**TL;DR:** Use Google, read lots, practice lots. Utilize the community.

I am self taught, hobby since early 2000's, took programming more seriously around 2012. I am not a book reader, I prefer to learn via online resources(blog posts, stack overflow, docs, communities like reddit). Usually I will have a project of interest to motivate me and figure out what I don't know as I go. Google search, read, if still can't understand, ask questions to communities and show what you have done so far to try and figure it out.

There is plenty of information out there to learn from in whatever medium works best for you. The key is putting in the time to practice what you learn, I advise smaller less complicated projects especially if you are new to coding. With Lua, I already had general programming skills and experience, I knew what I wanted and found the Lua equivalent or learned about how it did some things differently. I started with modding PD2, first using others code and tweaking it to do what I want, unerstanding it and making smaller mods. BigLobby is not something I advise as first project, but adding BLT 2.0 support should be something you can do(follow the docs on it, look at how other mods migrated to BLT 2.0 with their Git commit history).

You will find the code in this repo is fairly clean already and well commented/structured. While what it does is not as simple as other mods, I know for a fact that it's codebase is in a good condition for contributors vs what you'd usually come across. This mod took me over 3 months just to get to a working stage, I was fairly competent at making PD2 mods and writing Lua, along with PD2 codebase familiarity. 

I don't see you being able to do much to justify a "standalone remastered" version, but you're welcome to fork the repo, it's open-source MIT license, so do what you like :)

**Comment by polarathene** [comment]
Please try the latest [2.47](https://github.com/polarathene/biglobby/releases/tag/v2.47) release, it has been updated to the latest U168.1 game update, thanks to a community member who sent in an update for the past few months of changes in the [2.46](https://github.com/polarathene/biglobby/releases/tag/v2.46) release. As long as everything is working, they are deserving of claiming the bounty for this issue.

**Comment by polarathene** [comment]
[Latest release link](https://github.com/polarathene/biglobby/releases/latest)(always points to latest release, currently v2.48). Feel free to try. 

Assuming updates from steam-test1 are working for everyone, the developer does not appear to be interested in claiming the bounty? Closing this issue.
